cmd/neofs-adm/internal/modules/morph/initialize_deploy.go

```diff
@@ -217,7 +221,12 @@ func (c *initializeContext) deployContracts(method string) error {
 		if method == updateMethodName {
 			ctrHash, err = nnsResolveHash(c.Client, nnsHash, ctrName+".neofs")
 			if err != nil {
-				return fmt.Errorf("can't resolve hash for contract update: %w", err)
+				if errors.Is(err, errMissingNNSRecord) {
+					// if contract not found we deploy it instead of update
+					method = deployMethodName
```

Comment:
We should add `method := method` to the start of this loop, otherwise `method` will be changed for the following iterations. `subnet` is last contract but we shouldn't depend on this.